<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1179</width>
    <height>642</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QGraphicsView" name="canvas"/>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <layout class="QGridLayout" name="gridLayout">
              <item row="0" column="0">
               <widget class="QLabel" name="label">
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>16777212</height>
                 </size>
                </property>
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                  <weight>75</weight>
                  <italic>false</italic>
                  <bold>true</bold>
                 </font>
                </property>
                <property name="text">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:18pt; font-weight:400;&quot;&gt;Алгоритм:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QComboBox" name="algorithms">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                  <weight>50</weight>
                  <italic>false</italic>
                  <bold>false</bold>
                  <underline>false</underline>
                  <strikeout>false</strikeout>
                  <kerning>true</kerning>
                 </font>
                </property>
                <property name="cursor">
                 <cursorShape>PointingHandCursor</cursorShape>
                </property>
                <property name="toolTipDuration">
                 <number>-2</number>
                </property>
                <property name="inputMethodHints">
                 <set>Qt::ImhNone</set>
                </property>
                <property name="editable">
                 <bool>false</bool>
                </property>
                <property name="duplicatesEnabled">
                 <bool>false</bool>
                </property>
                <item>
                 <property name="text">
                  <string>Каноническое уравнение</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Параметрическое уравнение</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Алгоритм Брезенхема</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Алгоритм средней точки</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Библиотечный</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="1" column="0">
               <widget class="QLabel" name="label_22">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                 </font>
                </property>
                <property name="text">
                 <string>Радиус:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1" colspan="2">
               <widget class="QSpinBox" name="r_spinbox">
                <property name="minimum">
                 <number>0</number>
                </property>
                <property name="maximum">
                 <number>100000</number>
                </property>
               </widget>
              </item>
              <item row="1" column="3">
               <widget class="QLabel" name="label_21">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                 </font>
                </property>
                <property name="text">
                 <string>Полуось a:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="4" colspan="2">
               <widget class="QSpinBox" name="a_spinbox">
                <property name="minimum">
                 <number>0</number>
                </property>
                <property name="maximum">
                 <number>100000</number>
                </property>
                <property name="value">
                 <number>0</number>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_4">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                 </font>
                </property>
                <property name="text">
                 <string>Центр окружности:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QSpinBox" name="x0_circle_spinbox">
                <property name="minimum">
                 <number>-1000</number>
                </property>
                <property name="maximum">
                 <number>1000</number>
                </property>
                <property name="value">
                 <number>275</number>
                </property>
               </widget>
              </item>
              <item row="2" column="2">
               <widget class="QSpinBox" name="y0_circle_spinbox">
                <property name="minimum">
                 <number>-1000</number>
                </property>
                <property name="maximum">
                 <number>1000</number>
                </property>
                <property name="value">
                 <number>275</number>
                </property>
               </widget>
              </item>
              <item row="2" column="3">
               <widget class="QLabel" name="label_10">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                 </font>
                </property>
                <property name="text">
                 <string>Полуось b:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="4" colspan="2">
               <widget class="QSpinBox" name="b_spinbox">
                <property name="maximum">
                 <number>100000</number>
                </property>
               </widget>
              </item>
              <item row="3" column="3">
               <widget class="QLabel" name="label_5">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                 </font>
                </property>
                <property name="text">
                 <string>Центр эллипса:</string>
                </property>
               </widget>
              </item>
              <item row="3" column="4">
               <widget class="QSpinBox" name="x0_ellipse_spinbox">
                <property name="minimum">
                 <number>-1000</number>
                </property>
                <property name="maximum">
                 <number>1000</number>
                </property>
                <property name="value">
                 <number>275</number>
                </property>
               </widget>
              </item>
              <item row="3" column="5">
               <widget class="QSpinBox" name="y0_ellipse_spinbox">
                <property name="minimum">
                 <number>-1000</number>
                </property>
                <property name="maximum">
                 <number>1000</number>
                </property>
                <property name="value">
                 <number>275</number>
                </property>
               </widget>
              </item>
              <item row="3" column="0" colspan="3">
               <widget class="QPushButton" name="draw_circle_button">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                 </font>
                </property>
                <property name="cursor">
                 <cursorShape>PointingHandCursor</cursorShape>
                </property>
                <property name="text">
                 <string>Построить окружность</string>
                </property>
               </widget>
              </item>
              <item row="4" column="3" colspan="3">
               <widget class="QPushButton" name="draw_ellipse_button">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                 </font>
                </property>
                <property name="cursor">
                 <cursorShape>PointingHandCursor</cursorShape>
                </property>
                <property name="text">
                 <string>Построить эллипс</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0" colspan="3">
               <widget class="QLabel" name="label_23">
                <property name="text">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:14pt; font-weight:600;&quot;&gt;Построение окружности:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item row="0" column="3" colspan="3">
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:14pt; font-weight:600;&quot;&gt;Построение эллипса:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QGridLayout" name="gridLayout_3">
              <item row="2" column="3">
               <widget class="QSpinBox" name="b0_sp">
                <property name="maximum">
                 <number>1000</number>
                </property>
               </widget>
              </item>
              <item row="0" column="0" colspan="2">
               <widget class="QLabel" name="label_24">
                <property name="text">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:14pt; font-weight:600;&quot;&gt;Построение окружностей:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item row="0" column="2" colspan="2">
               <widget class="QLabel" name="label_11">
                <property name="text">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:14pt; font-weight:600;&quot;&gt;Построение эллипсов:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_19">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                 </font>
                </property>
                <property name="text">
                 <string>Начальный радиус:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QSpinBox" name="rn_spinbox">
                <property name="minimum">
                 <number>0</number>
                </property>
                <property name="maximum">
                 <number>100000</number>
                </property>
                <property name="value">
                 <number>0</number>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="label_18">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                 </font>
                </property>
                <property name="text">
                 <string>Шаг для радиуса:</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QSpinBox" name="hr_spinbox">
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>100000</number>
                </property>
               </widget>
              </item>
              <item row="3" column="2">
               <widget class="QLabel" name="label_13">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                 </font>
                </property>
                <property name="text">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Шаг для большей полуоси:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item row="3" column="3">
               <widget class="QSpinBox" name="ha_spinbox">
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>100000</number>
                </property>
               </widget>
              </item>
              <item row="4" column="0">
               <widget class="QLabel" name="label_17">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                 </font>
                </property>
                <property name="text">
                 <string>Количество окружностей:</string>
                </property>
               </widget>
              </item>
              <item row="4" column="1">
               <widget class="QSpinBox" name="c_circle_spinbox">
                <property name="maximum">
                 <number>100000</number>
                </property>
               </widget>
              </item>
              <item row="4" column="2">
               <widget class="QLabel" name="label_16">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                 </font>
                </property>
                <property name="text">
                 <string>Количество эллипсов:</string>
                </property>
               </widget>
              </item>
              <item row="4" column="3">
               <widget class="QSpinBox" name="c_ellipse_spinbox">
                <property name="maximum">
                 <number>100000</number>
                </property>
               </widget>
              </item>
              <item row="5" column="0" colspan="2">
               <widget class="QPushButton" name="draw_conc_circles_button">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                 </font>
                </property>
                <property name="cursor">
                 <cursorShape>PointingHandCursor</cursorShape>
                </property>
                <property name="text">
                 <string>Построить спектр конц. окружностей</string>
                </property>
               </widget>
              </item>
              <item row="5" column="2" colspan="2">
               <widget class="QPushButton" name="draw_conc_ellipses_button">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                 </font>
                </property>
                <property name="cursor">
                 <cursorShape>PointingHandCursor</cursorShape>
                </property>
                <property name="text">
                 <string>Построить спектр конц. эллипсов</string>
                </property>
               </widget>
              </item>
              <item row="1" column="3">
               <widget class="QSpinBox" name="a0_sp">
                <property name="maximum">
                 <number>1000</number>
                </property>
               </widget>
              </item>
              <item row="1" column="2">
               <widget class="QLabel" name="label_7">
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>16777213</height>
                 </size>
                </property>
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                  <underline>false</underline>
                 </font>
                </property>
                <property name="text">
                 <string>Начальная полуось а:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="2">
               <widget class="QLabel" name="label_8">
                <property name="font">
                 <font>
                  <family>Apple Braille</family>
                 </font>
                </property>
                <property name="mouseTracking">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Начальная полуось b:</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="3">
             <widget class="QPushButton" name="color_ellipse_button">
              <property name="maximumSize">
               <size>
                <width>400</width>
                <height>32</height>
               </size>
              </property>
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QPushButton" name="color_circle_button">
              <property name="font">
               <font>
                <family>Apple Braille</family>
                <weight>50</weight>
                <bold>false</bold>
               </font>
              </property>
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="autoFillBackground">
               <bool>false</bool>
              </property>
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="label_3">
              <property name="font">
               <font>
                <family>Apple Braille</family>
               </font>
              </property>
              <property name="text">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Цвет эллипса:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_2">
              <property name="font">
               <font>
                <family>Apple Braille</family>
               </font>
              </property>
              <property name="text">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Цвет окружности:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0" colspan="4">
             <widget class="Line" name="line">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_6">
              <property name="font">
               <font>
                <family>Apple Braille</family>
               </font>
              </property>
              <property name="text">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Цвет фона:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QPushButton" name="color_bg_button">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item row="2" column="2" colspan="2">
             <widget class="QComboBox" name="how_draw">
              <item>
               <property name="text">
                <string>Рисовать выбранным цветом</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Рисовать цветом фона</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="clear_canvas_button">
          <property name="font">
           <font>
            <family>Apple Braille</family>
           </font>
          </property>
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="text">
           <string>Очистить поле для рисования</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1179</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>r_spinbox</tabstop>
  <tabstop>x0_circle_spinbox</tabstop>
  <tabstop>y0_circle_spinbox</tabstop>
  <tabstop>draw_circle_button</tabstop>
  <tabstop>rn_spinbox</tabstop>
  <tabstop>hr_spinbox</tabstop>
  <tabstop>c_circle_spinbox</tabstop>
  <tabstop>draw_conc_circles_button</tabstop>
  <tabstop>a_spinbox</tabstop>
  <tabstop>b_spinbox</tabstop>
  <tabstop>x0_ellipse_spinbox</tabstop>
  <tabstop>y0_ellipse_spinbox</tabstop>
  <tabstop>draw_ellipse_button</tabstop>
  <tabstop>a0_sp</tabstop>
  <tabstop>b0_sp</tabstop>
  <tabstop>ha_spinbox</tabstop>
  <tabstop>c_ellipse_spinbox</tabstop>
  <tabstop>draw_conc_ellipses_button</tabstop>
  <tabstop>color_circle_button</tabstop>
  <tabstop>color_ellipse_button</tabstop>
  <tabstop>color_bg_button</tabstop>
  <tabstop>how_draw</tabstop>
  <tabstop>clear_canvas_button</tabstop>
  <tabstop>canvas</tabstop>
  <tabstop>algorithms</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
